src/index.js

```diff
@@ -1,65 +1,55 @@
 // import { add } from 'lodash';
 
+// import { forEach } from 'lodash';
 import './style.css';
 
-// This code is based on the video: https://www.youtube.com/watch?v=ePzOFu2xXUQ
+import { updateLocalStorage, toDoTasks } from './modules/class.js';
+
+// This code is based on the videos: https://www.youtube.com/watch?v=ePzOFu2xXUQ and https://www.youtube.com/watch?v=MkESyVB4oUw&t=1904s
 // and adapted to the Microverse requirements.
 
 const formList = document.querySelector('.form');
 const addNew = document.querySelector('.input');
 const container = document.querySelector('.list');
 const clearAll = document.querySelector('.clear');
 const upDate = document.querySelector('.refresh');
-let list = JSON.parse(localStorage.getItem('list')) || [];
-
-class ToDoTasks {
-  constructor(description = '', completed = false, index = null) {
-    this.description = description;
-    this.completed = completed;
-    this.index = index;
-  }
-}
-
-const toDoTasks = new ToDoTasks();
-
-const updateLocalStorage = () => {
-  const newLists = document.querySelectorAll('li');
-  list = [];
-  newLists.forEach((newList) => {
-    list.push({
-      description: newList.innerText,
-      completed: newList.classList.contains('checked'),
-      index: list.length + 1,
-    });
-  });
-  localStorage.setItem('list', JSON.stringify(list));
-};
 
 const toDoList = (task) => {
-  let newTask = addNew.value;
+  toDoTasks.description = addNew.value;
+  const newTask = toDoTasks.description;
   if (newTask.length < 1) return;
-  if (task) {
-    newTask = toDoTasks.description;
-  }
 
   const newList = document.createElement('li');
   if (task && task.completed) {
     newList.classList.add('checked');
   }
-  newList.innerText = newTask;
+  const taskInput = document.createElement('input');
+  taskInput.classList.add('text');
+  taskInput.type = 'text';
+  taskInput.value = toDoTasks.description;
+  taskInput.setAttribute('readonly', 'readonly');
+  newList.appendChild(taskInput);
   addNew.value = '';
+  container.appendChild(newList);
+
   const checkBtnEl = document.createElement('div');
   checkBtnEl.innerHTML = `
   <i class="fas fa-check-square"></i>
   `;
   newList.appendChild(checkBtnEl);
-  container.appendChild(newList);
+
   const trashBtnEl = document.createElement('div');
   trashBtnEl.innerHTML = `
   <i class="fas fa-trash"></i>
   `;
   newList.appendChild(trashBtnEl);
 
+  const editBtnEl = document.createElement('div');
+  editBtnEl.innerHTML = `
+  <i class="fas fa-solid fa-ellipsis-vertical"></i>
+  `;
+  newList.appendChild(editBtnEl);
+
   checkBtnEl.addEventListener('click', () => {
     newList.classList.toggle('checked');
     updateLocalStorage();
```

Comment:
- [ ] After editing a task, the updated task is not saved in the local storage. All changes to the To-Do List should be saved in the local storage. Please let's fix this issue.

- [ ] Also make sure that deleting a task should update all the remaining item's indexes so they represent the current list order and are unique. In the end, all changes should be saved in the local storage.

Reply:
Done.